lib/services/slack_command_service.ex

```diff
@@ -28,19 +28,13 @@ defmodule Aprb.Service.SlackCommandService do
           |> Enum.join(" ")
         "Subscribed topics: #{current_topics}"
 
-      Regex.match?( ~r/unsubscribe/ , params[:text])  ->
+      params[:text] =~ ~r/unsubscribe/ ->
         [_command | topic_names] = String.split(params[:text], ~r{\s}, parts: 2)
         # add subscriptions
-        removed_topics = for topic_name <- List.first(topic_names) |> String.split(~r{\s}) do
-          topic = Repo.get_by(Topic, name: topic_name)
-          if topic do
-            subscription = Repo.get_by(Subscription, subscriber_id: subscriber.id, topic_id: topic.id)
-            if subscription do
-              Repo.delete(subscription)
-              topic_name
-            end
-          end
-        end
+        removed_topics =
+          List.first(topic_names)
+            |> String.split(~r{\s})
+            |> Enum.map(fn(topic_name) -> unsubscribe(subscriber, topic_name) end)
```

Comment:
Nice refactors 👍, good idea to extract these into their own methods. These are very readable now. Wonder if `post_message` could benefit from something similar?

Reply:
Yes post message is still little hard to read, im planning to do another round of refactor and also switch to use new Elixir's formatting thing which should provide some guidance on how to improve this.